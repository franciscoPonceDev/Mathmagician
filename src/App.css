.App {
  text-align: center;
}

header {
  width: 100%;
  display: flex;
  justify-content: space-between;
}

nav ul {
  width: 100%;
  display: flex;
  justify-content: space-evenly;
  align-items: center;
  list-style: none;
}
