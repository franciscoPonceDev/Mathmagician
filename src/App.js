import React from 'react';
import './App.css';
import { BrowserRouter, Routes, Route } from 'react-router-dom';
import Calculator from './components/Calculator';
import Home from './components/Home';
import Header from './components/Header';
import Quote from './components/Quote';

const App = () => (
  <BrowserRouter>
    <Header />
    <Routes>
      <Route path="/" element={<Home />} />
      <Route path="/calculator" element={<Calculator />} />
      <Route path="/quote" element={<Quote />} />
    </Routes>
  </BrowserRouter>
);

export default App;
