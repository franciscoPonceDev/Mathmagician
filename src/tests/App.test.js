/* eslint-disable no-undef */
/* eslint-disable no-restricted-globals */

import React from 'react';
import renderer from 'react-test-renderer';
import {
  render,
  fireEvent,
  screen,
} from '@testing-library/react';
import App from '../App';
import '@testing-library/jest-dom/extend-expect';

it('App Renders Correctly', () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Clicking on Calculator link renders Calculator', () => {
  render(<App />);
  fireEvent.click(screen.getByText('Calculator'));
  expect(screen.getByText('Do some math!')).toBeInTheDocument();
});

it('Clicking on Home link renders home', () => {
  render(<App />);
  fireEvent.click(screen.getByText('Home'));
  expect(screen.getByText('Welcome to Mathmagician')).toBeInTheDocument();
});

it('Clicking on Quote link renders Quote', () => {
  render(<App />);
  fireEvent.click(screen.getByText('Quote'));
  expect(screen.getByText('Mathemagics is not about numbers, equations,computations, or algorithms; Is about understanding. -William Paul Thurston')).toBeInTheDocument();
});
