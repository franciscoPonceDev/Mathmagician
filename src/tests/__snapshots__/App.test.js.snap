// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Renders Correctly 1`] = `
Array [
  <header>
    <h1>
      Math Magician
    </h1>
    <nav
      className="nav"
    >
      <ul>
        <li>
          <a
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </li>
        <li>
          <p>
            |
          </p>
        </li>
        <li>
          <a
            href="/calculator"
            onClick={[Function]}
          >
            Calculator
          </a>
        </li>
        <li>
          <p>
            |
          </p>
        </li>
        <li>
          <a
            href="/quote"
            onClick={[Function]}
          >
            Quote
          </a>
        </li>
      </ul>
    </nav>
  </header>,
  <section
    className="home"
  >
    <h1>
      Welcome to Mathmagician
    </h1>
    <p>
      Id enim mollit ipsum nostrud. Dolor ipsum quis laborum quis amet adipisicing dolore excepteur do elit excepteur ea. Magna occaecat consequat anim aliquip Lorem dolore non sunt dolore. Nulla incididunt ex aute reprehenderit nisi consectetur nostrud.
    </p>
    <p>
      Commodo velit Lorem voluptate commodo nulla nulla adipisicing nulla exercitation ea. Do officia fugiat tempor ullamco exercitation nulla dolor excepteur labore in elit nostrud adipisicing. Sunt sunt laboris ea ut irure labore labore occaecat labore. Est laborum sint minim tempor magna reprehenderit laboris aliqua amet. Sint pariatur anim aute Lorem aute cillum sint proident. Eu exercitation ea culpa ad irure ut consequat reprehenderit amet ullamco consequat esse. Occaecat excepteur fugiat nulla ipsum cupidatat aliqua excepteur adipisicing voluptate voluptate adipisicing non irure officia.
    </p>
  </section>,
]
`;
